packages/edit-site/src/hooks/commands/use-wp-admin-commands.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis } from '@wordpress/commands';
+import { __, sprintf } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import { unlock } from '../../private-apis';
+
+const { useCommandLoader } = unlock( privateApis );
+
+const getWPAdminCreateCommandLoader = ( postType ) =>
+	function useCreateCommandLoader( { search } ) {
+		let label;
+		if ( postType === 'post' ) {
+			label = __( 'Create a new post' );
+		} else if ( postType === 'page' ) {
+			label = __( 'Create a new page' );
+		} else {
+			throw 'unsupported post type ' + postType;
+		}
+		const hasRecordTitle =
+			!! search && ! label.toLowerCase().includes( search.toLowerCase() );
+		if ( postType === 'post' && hasRecordTitle ) {
+			/* translators: %s: Post title placeholder */
+			label = sprintf( __( 'Create a new post "%s"' ), search );
+		} else if ( postType === 'page' && hasRecordTitle ) {
+			/* translators: %s: Page title placeholder */
+			label = sprintf( __( 'Create a new page "%s"' ), search );
+		}
+
+		const newPostLink = useSelect( ( select ) => {
+			select( editSiteStore ).getEditedPostType();
+			const { getSettings } = unlock( select( editSiteStore ) );
+			return getSettings().newPostLink ?? 'post-new.php';
+		}, [] );
+
+		const commands = useMemo(
+			() => [
+				{
+					name: 'core/wp-admin/create-' + postType,
+					label,
+					callback: () => {
+						document.location.href = addQueryArgs( newPostLink, {
+							post_type: postType,
+							post_title: hasRecordTitle ? search : undefined,
+						} );
+					},
+				},
+			],
+			[ newPostLink, hasRecordTitle, search, label ]
+		);
+
+		return {
+			isLoading: false,
+			commands,
+		};
+	};
+
+const useCreatePostLoader = getWPAdminCreateCommandLoader( 'post' );
+const useCreatePageLoader = getWPAdminCreateCommandLoader( 'page' );
+
+export function useWPAdminCommands() {
```

Comment:
I'm wondering if these commands should be part of the commands package. Besides the `newPostLink` setting(that I think it doesn't exist and I believe is not related to the site editor) and `getEditedPostType`(that is not used) I don't think we need it here and we could reuse it in other editors.

Reply:
I don't think so, the commands package is generic and independent of WP, think of it as an abstraction on top of paco's API to allow dynamic commands and all.

I do agree though, that most of the commands will be shared across screens and that we'd need a "core-commands" package at some point.